### ISSUE TITLE
`assert_type` rejects concrete function types asserted as equivalent `Callable[...]`

### ISSUE THREAD

**Issue by 4anklee** [astral-sh/ty#3435]
### Summary

ty currently rejects `assert_type` when the inferred type is a concrete function type and the asserted type is an equivalent `Callable[...]`.

```py
from collections.abc import Awaitable, Callable
from typing_extensions import assert_type

def sync_func() -> str:
    return "x"

async def async_func(x: int) -> str:
    return str(x)

assert_type(sync_func, Callable[[], str])
assert_type(async_func, Callable[[int], Awaitable[str]])
```

### Actual behavior

ty reports that the inferred function type is a subtype, or an unspellable concrete function type, but not equivalent to the asserted Callable[...].

### Expected behavior
These assertions should pass, matching pyright and mypy.

I understand that assert_type checks exact type equivalence, but for function values it would be useful to normalize concrete function types to their callable signatures when the asserted type is Callable[...].

### Version

ty 0.0.34

**Comment by AlexWaygood** [comment]
This is intentional behaviour. `assert_type` should only pass if the two types are _equivalent_ types. Here, ty infers a more precise type for the function than can be expressed using `Callable[...]` in type annotations.

Note that the error code emitted here is `assert-type-unspellable-subtype`, not `type-assertion-failure`. `assert-type-unspellable-subtype` is a precise error code for specifically this situation, where we can see that the inferred type is a subtype of the asserted type, _and_ we can also see that it's actually impossible to spell the inferred type using type annotations currently. If the inferred type were spellable, we would emit `type-assertion-failure` here instead of `assert-type-unspellable-subtype`. If you find this kind of diagnostic too noisy, you can switch off `assert-type-unspellable-subtype` in your ty config, and you'll still get `type-assertion-failure` diagnostics for the "more serious" errors.

https://docs.astral.sh/ty/reference/rules/#assert-type-unspellable-subtype

**Comment by 4anklee** [comment]
Thank you so much for the clarification! I will close this.
